metadata.yaml

```diff
@@ -27,9 +27,8 @@ containers:
 resources:
   alertmanager-image:
     type: oci-image
-    # The officially supported image is ubuntu/prometheus-alertmanager
     description: OCI image for alertmanager
-
+    upstream-source: ubuntu/prometheus-alertmanager
 provides:
   alerting:
     # The provider (alertmanager) adds the following key-value pair to the relation data bag of
```

Comment:
You may want to check out this library, that makes adding a schema to relations like this very simple:

https://github.com/canonical/serialized-data-interface

Then we don't have to have comments like this, as the schemas are quite readable on their own. As an example, here's the schema for the `ingress` relation used in Kubeflow:

https://github.com/canonical/operator-schemas/blob/master/ingress.yaml

Reply:
I really like the schema idea.
Going all-in with SDI would need a PR of its own.
I also heard that relation schemas are coming soon from juju itself..?

Reply:
I'm not sure that what is on the roadmap matches what SDI provides, though @jnsgruk can speak more to that

Reply:
My understanding is this:

- We have some underlying work to do in Juju core to enable some sort of execution environment, which would indeed enable validation/enforcement of schemas (this is the Lua stuff you've heard about)
- Once that is in place, we will develop it further to enable schema validation on config, relation databags, etc. (implied here is more complex, nested config/databags)

I think in light of this, where SDI is already in use, why not?! It works, it solves a problem for you guys. For O11y, it may just be prudent to wait out, but I'm certainly open to the discussion.